### PR TITLE
Added new tests/testfuncs/build_jar.sh file, used to build .jar files

### PR DESCRIPTION
**Pull request by johnTcrawford**
for both the "normal" and "alternative" versions of
UserDefinedTestFunctions.java; and deleted the "canned" (pre-built)
version of testfuncs_alternative.jar (& its README), per Chris's review
comments.

Adjusted tests/sqlcmd/build_jar.sh to call the new one above, rather
than build these .jar files itself (or use a "canned" version); and made
tiny adjustments to 7 sqlcmdtest .in and .outbaseline files to load
these .jar files in their new location.

Added a new .gitignore file in tests/testfuncs/, and adjusted the one in
tests/sqlcmd/, to ignore all relevant .jar files (& other files, such as
.errdiffs & .outdiffs); and removed redundant entries from the main
<voltdb>/.gitignore file.

UserDefinedTestFunctions.java: just changed USE_ALTERNATIVE_VERSION to
be 'final'; and deleted the duplicate "copy" (actually a symbolic link)
under tests/sqlcmd/org/voltdb_testfuncs/.

### CODE REVIEW

File: tests/sqlcmd/functions/org/voltdb_testfuncs/UserDefinedTestFunctions.java [hunk 1]
```diff
@@ -1,1116 +0,0 @@
-/* This file is part of VoltDB.
```
[the remark this answers is not in the document]
Comment (wolffcm) [reply tests/sqlcmd/functions/org/voltdb_testfuncs/UserDefinedTestFunctions.java]: Maybe it's just how delete files show up in the diff on GitHub.

File: tests/testfuncs/build_jar.sh [hunk 2]
```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+#
+# Compiles the UserDefinedTestFunctions class, which contains multiple UDFs
+# (user-defined (Java) functions), and creates two different jar files
+# containing it, which are used to test UDFs. One of the jar files is an
+# "alternative" version, with USE_ALTERNATIVE_VERSION set to 'true'; the other
+# is a "normal" version, with USE_ALTERNATIVE_VERSION set to 'false'.
+
+help() {
+  echo 'Usage: ./build_jar.sh [--build=BUILD]'
```
Comment (wolffcm) [inline tests/testfuncs/build_jar.sh]: Maybe rename this to `build_udf_jar.sh`?  I was getting it confused with the other script of the same name.

File: tests/sqlcmd/functions/org/voltdb_testfuncs/UserDefinedTestFunctions.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (johnTcrawford) [reply tests/sqlcmd/functions/org/voltdb_testfuncs/UserDefinedTestFunctions.java]: Yes, it's being deleted (the README.md file too). I agree it's unclear in GitHub.

File: tests/testfuncs/org/voltdb_testfuncs/UserDefinedTestFunctions.java [hunk 3]
```diff
@@ -39,7 +39,7 @@
     // Change this to true, to get a very altered version of most of these
     // UDF's (specifically, the ones with "add" or "concat" in their names);
     // useful for testing of @UpdateClasses (see ENG-12856)
-    private boolean USE_ALTERNATIVE_VERSION = false;
+    private final boolean USE_ALTERNATIVE_VERSION = false;
```
[the remark this answers is not in the document]
Comment (johnTcrawford) [reply tests/testfuncs/org/voltdb_testfuncs/UserDefinedTestFunctions.java]: Done.

File: tests/testfuncs/build_jar.sh (hunk 2 above)
Comment (johnTcrawford) [reply tests/testfuncs/build_jar.sh]: Done.

**Pull request merged**